### DISCUSSION TITLE
Multi-site/domains in one project

### DISCUSSION THREAD

**Discussion by bawpcwpn** [vercel/next.js#50369] [Help]
### Summary

Hi! Looking for suggestions on how to achieve running multiple sites from the one Next.js project, split by path.

We have a project that ideally we'd love to have in the one project, using the new app router in v13. 
We'd like to have one site at `auth.mydomain.com` point to `/app/auth` routes, and another eg. `upa.mydomain.com` point to `/app/upa` routes. 

Ideally we could still develop it from `localhost:3000/auth/` and `localhost:3000/upa` with nothing in the root, and it would deploy to one location. 

On the server, have considered using a proxy pass to direct requests to each path, the issue then is fixing the links for production and not development. A helper function could assist here.

Mainly wondering if this seems like the best way to achieve this, or if there are better ways to handle this kind of project with Next?

Thank you in advance for any suggestions!

**Comment by devjiwonchoi** [comment]
Right, it seems like your case is the best option for your requirements. 
If I were you, I would have set the load balancer's rule for each path `/upa` and `/auth` and limited the accessibility, 
in which you do not need to set two different subdomains for each path. 

I'm sure you're aware of leaving out the "root" `page.tsx` which leads to your `/` path is highly discouraged to do so.

In summary, your choice seems to be the best option for your current expectation.

**Comment by tj-noor** [comment]
You can achieve this by using a middleware - take a look at the Platforms example: https://github.com/vercel/platforms

To summarize:

1. in your next `middleware.ts|js` file, extract the hostname & pathname from the request, e.g.
   - `request.headers.get("host")`
   - `request.nextUrl.pathname`

2. parse through the hostname to identify what `site` it should go to

3. return `NextResponse.rewrite` with the respective route in the format: `/${site}${request.nextUrl.pathname}`. Do not include `/app` or `/pages` as Next handles that internally.

This way, you can go to `auth.mydomain.com` or `auth.localhost:3000` and it will render `/app|pages/auth/....`.

And wherever you are hosting, add the respective subdomains to resolve accordingly. You may also have to update your local dns to resolve to `(site).localhost:3000`.

**Reply by biapar** [reply]
How to with domain www.mydomain.com/aaaa and www.mydomain.com/bbbb ?
